### DISCUSSION TITLE
[Implemented] Revert to original window size and position from maximize (toggle maximize)

### DISCUSSION THREAD

**Discussion by AObuchow** [rxhanson/Rectangle#894] [Ideas]
I'm coming to macOS (and rectangle) after using KDE Plasma for years on Linux. 
I really love Rectangle Pro so far, but there's a feature I wish it had:

When you maximize a window with the shortcut, it would be really nice if you could repeat the same shortcut and revert the window back to its original position and size, prior to maximizing. This way, when you have a number of windows on screen at once, you can quickly maximize one, then restore it back to its original position and size to restore your window layout.

Here's a demonstration of what this looks like on KDE Plasma (I'm using the same shortcut to toggle window maximize on/off) :
![rectangle_feature_request](https://user-images.githubusercontent.com/10300119/186794716-2f73d6e7-0789-48ad-96cf-cd96e0518661.gif)

I'd be willing to try and contribute this feature myself, though I have no previous experience in Swift (yet) and I'm not sure how difficult this would be to implement. 

Please let me know if this would be feasible to add, and whether it would make sense as a feature for Rectangle :) 

Thank you in advance.

**Comment by rxhanson** [comment]
Ah right, I keep forgetting about this one. I'm going to roll out an update to Pro soon with the latest from Rectangle, so I'll put it on my Todo for this release. I don't have a set date for my next Rectangle release, but will plan to put it there as well.

**Reply by AObuchow** [reply]
Thanks for the quick reply - that sounds awesome :) I'm really looking forward to having this feature, thank you :)

**Comment by AObuchow** [comment]
@rxhanson I downloaded the latest update of Rectangle Pro today and was very happy to see this feature is now implemented. Thank you so much! :)

**Reply by rxhanson** [reply]
Glad you found it! I forgot to put it in the release notes, and I didn't squeeze it into the latest version of Rectangle yet.
